Add detailed description of TVM initialization stage

novusnota opened the pull request:
This is the deliverable for [this footstep](https://github.com/ton-society/ton-footsteps/issues/196)

1. Added a new page(section) — TVM Initialization
2. Added it to the `sidebars.js` (side bar menu)
3. Updated TVM Overview page to point to the new one
4. Made small QoL changes to TVM Overview page

Gusarich commented:
LGTM.

The initial task from [Footstep](https://github.com/ton-society/ton-footsteps/issues/196) was to rewrite the whitepaper's content into simpler language to enhance its readability. However, I'm uncertain about how to further simplify or improve the clarity of the text, considering it's already was written in fairly direct and concise technical language in the whitepaper.

novusnota commented:
@Gusarich I mostly agree, but there was still room for improvement. And such, I did:
- Re-structure the original content to better suit TON docs
- Added extra comments and adjustments to the original text
- Expanded description of **c7** and **c5**, while making sure they conform to the modern TVM implementation and structures of `block.tlb`
- Added a room for **c7** extension due to the recent and ongoing [TVM Upgrade](https://docs.ton.org/learn/tvm-instructions/tvm-upgrade-2023-07)

reveloper commented:
@novusnota, Thank you for the contribution!
